File: 2nd Year/2nd Semester/OOS/assignment_1/q5/complex.java
class Complex {
	int a,b;
	Complex() {
		a = 0;
		b = 0;
	}
	Complex(int t) {
		a=t;
		b=0;
	}
	Complex(int x, int y) {
		a=x;
		b=y;
	}
	Complex add(Complex c) {
		Complex tmp = new Complex(a+c.a, b+c.b);
		return tmp;
	}
	Complex multiply(Complex c) {
		Complex tmp = new Complex(a*c.a-b*c.b, b*c.a+a*c.b);
		return tmp;
	}
	/* 3+2i, 4-2i */
	void disp() {
		System.out.println(a+" + "+b+"i");
	}
	public static void main(String args[]) {
		Complex o1 = new Complex(3,2);
		Complex o2 = new Complex(4,-2);
		Complex o3 = o1.add(o2);
		Complex o4 = o1.multiply(o2);
		System.out.println("Num 1 is \n3+2i");
		System.out.println("Num 2 is \n4-2i");
		System.out.println("Sum is");
		o3.disp();
		System.out.println("Product is");
		o4.disp();
	}
}

File: 2nd Year/1st Semester/OOP/assignment4/complex.cpp
#include <iostream>

using namespace std;

class Complex {
	double a, b;
	public:
		Complex (double x=0, double y=0) {
			a=x;
			b=y;
		}
		Complex operator+(Complex c) {
			return Complex(a+c.a, b+c.b);
		}
		Complex operator-(Complex c) {
			return Complex(a-c.a, b-c.b);
		}
		Complex operator*(Complex c) {
			return Complex(a*c.a-b*c.b, a*c.b+b*c.a);
		}
		Complex operator/(Complex c) {
			return Complex((a*c.a + b*c.b)/(c.a*c.a+c.b*c.b), (b*c.a-a*c.b)/(c.a*c.a+c.b*c.b));
		}
		Complex& operator=(Complex c) {
			a=c.a;
			b=c.b;
		}
		void disp() {
			cout<< a << " + " << b << " i" << endl;
		}
};
int main() 
{
Complex c1(2,3), c2(3,5), c3;
c3=c1+c2;
cout << "c1 + c2 = ";
c3.disp();
c3=c1-c2;
cout << "c1 - c2 = ";
c3.disp();
c3=c1*c2;
cout << "c1 * c2 = ";
c3.disp();
c3=c1/c2;
cout << "c1 / c2 = ";
c3.disp();
}

